lib/usermanager.js

```diff
@@ -340,36 +340,36 @@ exports = module.exports = {
   },
 
   /**
-    * handle a users unshared courses on delete
-  */
+   * Handle a users courses on delete
+   */
   handleUserCoursesOnDelete: function(userId, courseOption, done) {
     var pluginManager = pluginmanager.getManager();
     var plugins = pluginManager.getPlugins();
     var CourseContent = require(plugins.content.course.fullPath);
-
     var courseContent = new CourseContent();
-    courseContent.retrieve({ createdBy: userId, _isShared: false }, {}, function (err, results) {
-
-      switch (courseOption) {
-        case 'transfer':
-          var currentUser = this.getCurrentUser();
-          async.each(results, function(model, cb) {
-            courseContent.update({ _id: model._id }, { createdBy: currentUser._id }, cb);
-          }, done);
-          break;
-        case 'delete':
-          async.each(results, function(model, cb) {
-            courseContent.destroy({ _id: model._id }, false, cb);
-          }, done);
-          break;
-        case 'share':
-          async.each(results, function(model, cb) {
-            courseContent.update({ _id: model._id }, { _isShared: true }, cb);
-          }, done);
-          break;
-        default:
-          done(null);
-      }
+    if (courseOption === null) {
+      return done(null);
+    }
+    courseContent.retrieve({ createdBy: userId }, {}, function (err, results) {
+      var currentUser = this.getCurrentUser();
+      async.eachSeries(results, function(model, cb) {
+        // Transfer ownership of all courses to current user
```

Comment:
Suggest adding a comment linking to this PR so we have a reference to your user deletion matrix.